[Identity] New extensions for Hybrid cache and Prompt Create

travlos opened the pull request:
New additions

1. Add support for HybridCache
2. Support Duende Prompt create
3. Extension to register certificate for IIS on windows

travlos committed:
Add support for IIS certificate

Added `AddCertificateIIS` method for IIS/Windows to handle
certificate registration issues, conditionally compiled for
`NET9_0_OR_GREATER`.

travlos committed:
Add Hybrid cache support for identity

travlos committed:
Build errors

travlos committed:
Support Duende Prompt create

travlos committed:
Fix build

cleftheris commented on src/Indice.Features.Identity.Core/Cache/HybridCacheStore.cs:
```diff
@@ -0,0 +1,60 @@
+/// Provides a hybrid cache implementation for storing and retrieving objects of type <typeparamref name="T"/>.
+/// </summary>
+/// <typeparam name="T">The type of object to cache.</typeparam>
+public class HybridCacheStore<T> : ICache<T> where T : class
```
rename to `DuendeHybridCache`. Consider making internal

cleftheris commented on src/Indice.Features.Identity.UI/IdentityBuilderUIExtensions.cs:
```diff
@@ -56,6 +58,14 @@ public static IServiceCollection AddIdentityUI(this IServiceCollection services,
                 options.ValidReturnUrls.Add(url);
             }
         });
+#if NET9_0_OR_GREATER
```
consider using the `IPostConfigureOptions<MySettings>` interface with a class in order to get injection services like link builder. https://andrewlock.net/delaying-strongly-typed-options-configuration-using-postconfigure-in-asp-net-core/

travlos committed:
Rename HybridCacheStore to DuendeHybridCache

travlos committed:
Refactor IdentityServerOptions configuration

Replaced inline IdentityServerOptions configuration with a new
IdentityServerOptionsPostConfigure class for modularity and
maintainability. Added conditional compilation for .NET 9.0+
to ensure backward compatibility. Updated CreateAccountUrl
to use ToLowerInvariant for consistent formatting. Removed
unused namespace from IdentityUIOptions.cs.

cleftheris reviewed (approved):
🚀

cleftheris committed:
Merge branch 'develop' into feature/identity-additions
